### PR TITLE
Make PkgTemplates work on 1.0

### PR DESCRIPTION
**Pull request by christopher-dG**
This doesn't change much if any of the package functionality, it just makes it run and all the test pass. So it's basically creating packages for 0.6. Next up is to use `Pkg.generate` etc.  
One other change is with the default plugin file templates, there was a bug fixed in Mustache which lets us write much more clearly.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,17 +3,18 @@ os:
   - linux
   - osx
 julia:
-  - 0.6
```
[the remark this answers is not in the document]
Comment (iamed2) [reply .travis.yml]: (and also require 0.7 instead of 1.0)

File: .travis.yml [hunk 2]
```diff
@@ -3,17 +3,18 @@ os:
...
     - julia: nightly
   fast_finish: true
 notifications:
   email: false
-script:
-  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-  - julia -e 'Pkg.clone(pwd()); Pkg.test("PkgTemplates"; coverage=true)'
+before_script:
+  - git config --global user.name "Chris de Graaf"
+  - git config --global user.email "chrisadegraaf@gmail.com"
```
Comment (omus) [inline .travis.yml]: Why is this needed?

File: defaults/appveyor.yml [hunk 3]
```diff
@@ -36,8 +36,14 @@ build_script:
...
-  - C:\projects\julia\bin\julia -e "cd(Pkg.dir(\"{{PKGNAME}}\")); Pkg.add(\"Coverage\"); using Coverage; Codecov.submit(process_folder())"{{/CODECOV}}{{#COVERALLS}}
-  - C:\projects\julia\bin\julia -e "cd(Pkg.dir(\"{{PKGNAME}}\")); Pkg.add(\"Coverage\"); using Coverage; Coveralls.submit(process_folder())"{{/COVERALLS}}{{/AFTER}}
+after_test:
+{{#CODECOV}}
+  - C:\projects\julia\bin\julia -e "cd(Pkg.dir(\"{{PKGNAME}}\")); Pkg.add(\"Coverage\"); using Coverage; Codecov.submit(process_folder())"
+{{/CODECOV}}
+{{#COVERALLS}}
+  - C:\projects\julia\bin\julia -e "cd(Pkg.dir(\"{{PKGNAME}}\")); Pkg.add(\"Coverage\"); using Coverage; Coveralls.submit(process_folder())"
+{{/COVERALLS}}
+{{/AFTER}}
```
Comment (omus) [inline defaults/appveyor.yml]: The default appveyor.yml script should be now based upon the Appveyor.jl package.

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (christopher-dG) [reply .travis.yml]: 👍

File: defaults/appveyor.yml (hunk 3 above)
Comment (christopher-dG) [reply defaults/appveyor.yml]: Yep, I'm going to change this in the next PR which will update the real functionality.

File: .travis.yml (hunk 2 above)
Comment (christopher-dG) [reply .travis.yml]: I should add a comment explaining. When you `LibGit2.commit` without these options, it raises an exception, and Travis OSX and AppVeyor both don't have this preconfigured (interestingly, it works on Linux). 
(A couple of commits are made as part of package generation)

Comment (omus) [reply .travis.yml]: Ok, you may not want to expose your email address in the commit history. Up to you

Comment (christopher-dG) [reply .travis.yml]: I mean, it's all over the commit history but I will remove it from the CI files 😅. Editing that commit.
